WIP IRSS Certificates

fhelfer opened the pull request.

kunzeus committed:
[fix] zero input

mjansenDatabay committed:
Init/ErrorHandling/SOAP: Use `SOAP` error handler ALWAYS in SOAP context

This PR suggest to always use the `SOAP` error handler if
ILIAS is executed in a `SOAP` context. Currently, the `SOAP`
error handler is only used in `DEVMODE`.

After PR #7046 has been merged I thought about the error handling
again and did not find a reason why we only use this error handler
in `DEVMODE`.

mjansenDatabay committed:
Auth: Make `ilSessionTest` work

mjansenDatabay committed:
Auth: Handle race conditions when writing sessions

See: https://mantis.ilias.de/view.php?id=40631

(cherry picked from commit 7628256f42def3b4912d00257133b6bf338e3921)
(cherry picked from commit acb1cea9c12c2fdfc34739e4d0c5302703c3500e)

mjansenDatabay committed:
Init/PasswordAssistance: Trim form values

See: https://mantis.ilias.de/view.php?id=39869
(cherry picked from commit a420fea7263562992b05b226f44c33ae4c8392cb)

kergomard committed:
Auth: Fix Max Idle Time Readout

See: https://mantis.ilias.de/view.php?id=41119
(cherry picked from commit eb535f94bb0be5a366e7021946e70c5bf4fd3a36)

mjansenDatabay committed:
Utils/FileUtils: Reset umask after creating directories

See: https://mantis.ilias.de/view.php?id=41139

# Conflicts:
#	Services/Utilities/classes/class.ilUtil.php

lscharmer committed:
Refinery: Add transformation for escaping html/css/js/url's

fhelfer committed:
WIP initial Certificate IRSS

fhelfer committed:
fix composer install

nhaagen committed:
ComponentsRevision: fix WAC

mjansenDatabay commented on components/ILIAS/WebServices/RPC/lib/.gitignore:
```diff
@@ -1,3 +1,4 @@
 bin
-tmp
\ No newline at end of file
+tmp
+target
```
Please remove these changes.

iszmais committed:
DataCollection: authority request

chfsx committed:
Moved Tables in AdministrativeNotifications to UI\Table\Data

chfsx committed:
DMBS: Proposed Version range für MySQL/MariaDB

mjansenDatabay committed:
ErrorHandling: Check for `tpl` existence

corro committed:
Additional File Suffix List

# Conflicts:
#	Services/Utilities/classes/class.ilFileUtils.php
#	src/Filesystem/Security/Sanitizing/FilenameSanitizerImpl.php

# Conflicts:
#	src/Filesystem/Security/Sanitizing/FilenameSanitizerImpl.php

mjansenDatabay commented on components/ILIAS/Certificate/classes/File/Certificate/class.ilPdfGenerator.php:
```diff
@@ -97,12 +106,18 @@ private function createPDFScalar(ilUserCertificate $certificate): string
         $certificateContent = str_replace(
             ['[BACKGROUND_IMAGE]', '[CLIENT_WEB_DIR]'],
-            ['[CLIENT_WEB_DIR]' . $certificate->getBackgroundImagePath(), 'file://' . CLIENT_WEB_DIR],
+            ['[CLIENT_WEB_DIR]' . $background_src, 'file://' . CLIENT_WEB_DIR],
```
I am sure we'll have to use a (signed) HTTP path here and should not use `file://` anymore.

mjansenDatabay commented on components/ILIAS/Certificate/classes/Setup/class.ilCertificateDatabaseUpdateSteps.php:
```diff
@@ -77,4 +77,24 @@ public function step_5(): void
+    public function step_6(): void
+    {
+        // TODO: Certificate Image Migrations
+        $this->db->createTable('', [
```
Why do we need a new table?

mjansenDatabay commented on components/ILIAS/Certificate/classes/File/CertificateResourceHandler.php:
```diff
@@ -0,0 +1,62 @@
+        $this->certificate_template_repo = new ilCertificateTemplateDatabaseRepository($DIC->database());
+        $this->stakeholder = new ilCertificateTemplateStakeholder();
+        $this->irss = $DIC->resourceStorage();
+        $this->global_certificate_settings = new ilObjCertificateSettings();
```
Is it possible to use dependency injection instead?

mjansenDatabay commented on components/ILIAS/Certificate/tests/ilPdfGeneratorTest.php:
```diff
@@ -51,35 +53,49 @@ public function testGenerateSpecificCertificate(): void
             ->disableOriginalConstructor()
-            ->getMock();
+            ->getMock()
+        ;
```
Can we pu the `;` behind the closing `)` (everywhere)?

mjansenDatabay commented on components/ILIAS/Certificate/classes/class.ilObjCertificateSettingsGUI.php:
```diff
@@ -144,8 +153,10 @@ public function settings(): void
-            ilWACSignedPath::setTokenMaxLifetimeInSeconds(15);
-            $bgimage->setImage(ilWACSignedPath::signFile($this->object->getDefaultBackgroundImagePathWeb()));
+            $rid = $this->object->getBackgroundImageIdentification();
+            if ($rid !== null) {
```
Do we really need to check this, even if we already asked `$this->object->hasBackgroundImage()`?

mjansenDatabay commented on components/ILIAS/Certificate/classes/class.ilObjCertificateSettingsGUI.php:
```diff
@@ -28,8 +30,12 @@
+    protected ILIAS\FileUpload\FileUpload $upload;
+
+    /** @var ilObjCertificateSettings */
+    protected ?ilObject $object;
```
Can we make them `private`?

mjansenDatabay commented on components/ILIAS/Certificate/classes/User/class.ilUserCertificateGUI.php:
```diff
@@ -234,7 +241,8 @@ public function listCertificates(): void
                     ->icon()
-                    ->standard($certificateData['obj_type'], $certificateData['obj_type']);
+                    ->standard($certificateData['obj_type'], $certificateData['obj_type'])
+                ;
```
Please don't put a single `;` in its own line.

mjansenDatabay commented on components/ILIAS/Certificate/classes/Template/class.ilCertificateTemplateDatabaseRepository.php:
```diff
@@ -28,10 +28,12 @@ class ilCertificateTemplateDatabaseRepository implements ilCertificateTemplateRe
     private readonly ilLogger $logger;
     private readonly ilObjectDataCache $objectDataCache;
 
+    public const TABLE_NAME = 'il_cert_template_new';
```
Please move constants at the top of the declarations.

mjansenDatabay commented on components/ILIAS/Certificate/classes/User/class.ilUserCertificateRepository.php:
```diff
@@ -27,10 +27,12 @@ class ilUserCertificateRepository
     private readonly ilLogger $logger;
     private readonly string $defaultTitle;
 
+    public const TABLE_NAME = 'il_cert_user_cert_new';
```
Please move constants at the top of the declarations.

mjansenDatabay commented on components/ILIAS/Certificate/classes/Form/class.ilFormFieldParser.php:
```diff
@@ -91,7 +91,7 @@ public function fetchDefaultFormFields(string $content): array
         }
 
-        $xsl = file_get_contents("./components/ILIAS/Certificate/xml/fo2xhtml.xsl");
+        $xsl = file_get_contents("../components/ILIAS/Certificate/xml/fo2xhtml.xsl");
```
Maybe we should build the path based on `__DIR__`.

mjansenDatabay commented on components/ILIAS/Certificate/classes/Helper/ilCertificateDateHelper.php:
```diff
@@ -29,7 +29,7 @@ class ilCertificateDateHelper
     {
         if (null === $dateFormat) {
-            require_once 'components/ILIAS/Calendar/classes/class.ilDateTime.php'; // Required because of global contant IL_CAL_DATE
+            require_once './components/ILIAS/Calendar/classes/class.ilDateTime.php'; // Required because of global contant IL_CAL_DATE
```
Why not `../` like in your change above?

mjansenDatabay commented on components/ILIAS/Certificate/classes/Helper/ilCertificateDateHelper.php:
```diff
@@ -50,7 +50,7 @@ public function formatDate($date, ?int $dateFormat = null): string
     {
         if (null === $dateFormat) {
-            require_once 'components/ILIAS/Calendar/classes/class.ilDateTime.php'; // Required because of global contant IL_CAL_DATE
+            require_once './components/ILIAS/Calendar/classes/class.ilDateTime.php'; // Required because of global contant IL_CAL_DATE
```
Why not `../` like in your change above?

mjansenDatabay commented on components/ILIAS/Certificate/classes/Placeholder/Values/class.ilDefaultPlaceholderValues.php:
```diff
@@ -18,7 +18,7 @@
 declare(strict_types=1);
 
-require_once 'components/ILIAS/Calendar/classes/class.ilDateTime.php'; // Required because of global contant IL_CAL_DATE
+require_once './components/ILIAS/Calendar/classes/class.ilDateTime.php'; // Required because of global constant IL_CAL_DATE
```
Why not `../` like in your change above?

mjansenDatabay commented on components/ILIAS/Certificate/classes/File/ilCertificateTemplateStakeholder.php:
```diff
@@ -0,0 +1,26 @@
+<?php
+
```
Please add the license header.

mjansenDatabay commented on components/ILIAS/Certificate/classes/Template/Action/Preview/ilCertificateTemplatePreviewAction.php:
```diff
@@ -89,7 +96,8 @@ public function createPreviewPdf(int $objectId): void
         $pdf_base64 = $this->rpcClientFactoryHelper
-            ->ilFO2PDF('RPCTransformationHandler', $xlsfo);
+            ->ilFO2PDF('RPCTransformationHandler', $xlsfo)
+        ;
```
Please move the `;` one line up after the `)`;

mjansenDatabay commented on components/ILIAS/Certificate/classes/User/class.ilUserCertificateGUI.php:
```diff
@@ -196,21 +204,20 @@ public function listCertificates(): void
                 ->withLabel($this->language->txt($this->sortationOptions[$sorting]))
-                ->withTargetURL($this->ctrl->getLinkTarget($this, 'applySortation'), 'sort_by');
+                ->withTargetURL($this->ctrl->getLinkTarget($this, 'applySortation'), 'sort_by')
+            ;
```
Please move the `;` one line up after the `)`;

mjansenDatabay committed:
User/UDF: Don't provide NULL values for consumers

fneumann committed:
Fix #41360 - Privacy: Orphaned Forum Drafts (#7508)

(cherry picked from commit d97ad13e3d44a24442ee306e164a4d08f3167eb1)

klees committed:
Orgu: properly escape titles in org-unit tree

kergomard committed:
Object: Fix Import of Translation

See: https://mantis.ilias.de/view.php?id=41056

mjansenDatabay committed:
UI Tables: Fix type issues in tables (of my authorities)

mjansenDatabay committed:
Merge remote-tracking branch 'origin/trunk' into trunk

mBeym committed:
Sort table using DB query instead of in code

mBeym committed:
Fix missing tableview_id option parameter in DataCollection

lscharmer committed:
User: Add goto link for the login page

kergomard committed:
User: Relable User Menu in Metabar

See: https://mantis.ilias.de/view.php?id=32496

mjansenDatabay committed:
Forum: Null-coalesce for `gender`

(cherry picked from commit 663b49411a54919af445daecc6f66827a80fca3c)

kergomard committed:
Test: Fix Import With Reporting Date

nhaagen committed:
[FEATURE] UI: introduce new Table\Ordering component to change position of records. (#6290)

iszmais committed:
Transform all properties of symbols that are publicly accessible.This might have an effect on the presentation of names and labels of symbols (#7350)

kergomard committed:
UI: Fix In-Accessible Second Level Items Drilldown (#7454)

thibsy committed:
[FIX] UI: replace legacy paths from Icon unit tests.

fneumann committed:
[Forum] Remove ilObjForum::_deleteUser call from ilObjUser

The forum component already has an event listener for deleted user accounts.
The call of this function is moved there.
This allows a further internal refactoring in the forum component
and reduces the dependencies in ilObjUser.

alex40724 committed:
Trash: Reformat output of title and description (0041364) (#7512)

fneumann committed:
New Authorities for Administration Component

malaja11 committed:
error message for the upload of globally restricted file types by a normal user in FileUploadQuestion added

klees committed:
UI: fix backwards compatibility for Data/Table backport (#7521)

The method isClientSideValueOk changing from protected to public made plugins
break that implemented their own inputs. When switching back the method to public,
unit tests fail, indicating that we cross class hierarchies when calling the
method. This was due to the fact that we indeed had two different methods
isClientSideValueOk, one from the original input hierarchy, one from a trait
for groups. These are reconciled now and thus the method can be protected again.

matthiaskunkel committed:
Update maintenance.md

Added Oliver Samoila as new authority for Data Collection

aaronbidzan committed:
Test:  Info Message When No User Selected

fneumann committed:
Add company to the listed users

malaja11 committed:
threshold value in Skilllevelthresholds table is now required to be filled

klees committed:
Data: fix Meta/Tag::toHtml (#40110)

(cherry picked from commit b3d46ddec4d33c3f203f6ad84fd8643c6f724213)

fhelfer committed:
apply review certificates

klees committed:
IAss: remove duplicate 'no-entry'-message in table!

klees committed:
IAss: lead user to better tab on default

klees committed:
IAss: remove overly restrictive access check to lp-tab

mjansenDatabay committed:
TestQuestionPool: Ensure proper use of `unserialize` (#7528)

See: https://mantis.ilias.de/view.php?id=41368

mjansenDatabay committed:
Authentication: Prevent type casting of composed authentication modes

See: https://mantis.ilias.de/view.php?id=41381

fhelfer committed:
Move external Test Code out of Certificate

alex40724 committed:
disallow classes on certain unserialize calls

klees committed:
IAss: lead user to better tab on default (II)

kergomard committed:
Test: Fix Missing Page Editor

See: https://mantis.ilias.de/view.php?id=41358

alex40724 committed:
41220: Media Container adds white block to the bottom of images/videos

alex40724 committed:
37185: <- in remark or evaluation status is deleted

alex40724 committed:
36396: Export of ILIAS learning module to HTML with code fails

alex40724 committed:
34329: Subtitles in Video in Accordion not shown

alex40724 committed:
34329: Subtitles in Video in Accordion not shown

alex40724 committed:
37331: Metadaten an Seiten der ILIAS Lernmoduls nicht verfügbar

alex40724 committed:
36550: Resources of Other Users -> Show Shared Portfolio (with password)

alex40724 committed:
39539: Delete Item -> error

alex40724 committed:
40987: ILIAS Survey and Primary button

nhaagen committed:
TA 41390, hide user name in anon mode

iszmais committed:
Preload gui Data for Dashboard (#7523)

tfamula committed:
LUI: Replace ilModalGUI with UI Modals

mjansenDatabay committed:
ErrorHandling: Check for `tpl` existence

(cherry picked from commit a8519effc1c5d3f03a1891c1738d924805b65ba3)

mjansenDatabay committed:
Authentication: Use PHP core randomizer

mjansenDatabay committed:
Init: Use PHP core randomizer

pascalseeland committed:
Merge pull request #7542 from mjansenDatabay/improvement/trunk/auth-ilrandom

Authentication: Use PHP core randomizer

pascalseeland committed:
Merge pull request #7543 from mjansenDatabay/improvement/trunk/init-ilrandom

Init: Use PHP core randomizer

alex40724 committed:
41380: Cannot upload new images in content styles

alex40724 committed:
41365: Send Participation Confirmation crashes with error: Undefined array key 0

alex40724 committed:
41397: Missing notification when deleting export files in a Survey

fhelfer committed:
remove il prefix + snake case Certificates in Test

pascalseeland committed:
28365: Dcl - Add missing index

fhelfer committed:
remove use statements form global namspace Test Certificates

fhelfer committed:
fix namespacing for new Test Certificate namespace

pascalseeland committed:
41403: Dcl: Fix broken import

fhelfer committed:
Bugfix userLanguage undefined Cron save UserCertificate

fhelfer committed:
remove Test use in CMiXapi Certificate

mjansenDatabay committed:
AccessControl: Fix PHP type error for starting point role id

pascalseeland committed:
Fix 40882: Dcl Add missing type cast

kergomard committed:
Test: Fix Missing Parameter in Function

mBeym committed:
Fix trying to assign string value to int|null property

mBeym committed:
Fix mail template loaded from wrong directory

(cherry picked from commit 0ed2a70756b1eee18d6e210d3a0064ed824430ca)

iszmais committed:
Prevent division error on 0 max points (#7558)

alex40724 committed:
41377: Error When Tagging Material in ILIAS Session

chfsx committed:
Removed legacy table

chfsx committed:
Moved Tables in Bibliografic to UI\Table\Data

kergomard committed:
Test: Fix IP-Range on FixedParticipants

kergomard committed:
Test: Fix Test Startable Without Read Rights

kergomard committed:
Test: Fix Test Link in ObjectListAsCard

chfsx committed:
[FIX] deactivated memcached test on CI due to missing memcached server

kergomard committed:
Object: Set Online/Offline Correctly on Clone

See: https://mantis.ilias.de/view.php?id=41234

kergomard committed:
User: Remove Backlink from Profile

See: https://mantis.ilias.de/view.php?id=25668#c62031

tfamula committed:
Accessibility: Fix updating documents

chfsx committed:
[FIX] Split up test since second assertion never run

chfsx committed:
[FIX] shibboleth login

fhelfer committed:
Update Password Form in UserService to KS Components (ILIAS 10) (#7370)

fhelfer committed:
Add esscaping for Certificate ObjectSpecificPH

(cherry picked from commit 0fb23d859eba9452daaf6acbcb893fc8aea955a9)
(cherry picked from commit fd23678d3d8daf36a4a3cc2b608c69df894b0633)

chfsx committed:
[FIX] 0041435: Menü-Eintrag "Baumansicht" innerhalb von Administration führt zu Fehler

# Conflicts:
#	src/GlobalScreen/Scope/MainMenu/Collector/Renderer/TopParentItemDrilldownRenderer.php

fhelfer committed:
needed changes to make certificates work remove later

fhelfer committed:
some rebase bugfixes

fhelfer committed:
IRSS Migration

fhelfer committed:
IRSS Certificates Migration Part 2

fhelfer committed:
Remove target dir from gitignore
